src/foam/nanos/notification/EmailSetting.js

```diff
@@ -11,14 +11,47 @@ foam.CLASS({
   extends: 'foam.nanos.notification.NotificationSetting',
 
   javaImports: [
+    'foam.core.PropertyInfo',
+    'foam.dao.DAO',
+    'foam.nanos.auth.User',
     'foam.nanos.logger.Logger',
     'foam.nanos.notification.email.EmailMessage',
     'foam.util.Emails.EmailsUtility',
     'java.util.Arrays',
+    'java.util.Iterator',
     'java.util.List',
+    'java.util.Map',
+    'static foam.mlang.MLang.EQ'
   ],
 
   methods: [
+    {
+      name: 'mapNotificationArguments',
+      type: 'Map',
+      documentation: `Iterate through arguments to replace propertyInfo values with the notified user' values.`,
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'arguments', type: 'Map' },
+        { name: 'user', type: 'User' }
+      ],
+      javaCode: `
+        Logger logger = (Logger) x.get("logger");
+
+        Iterator entries = arguments.entrySet().iterator();
+        while (entries.hasNext()) {
+          Map.Entry entry = (Map.Entry) entries.next();
+          if ( entry.getValue() instanceof PropertyInfo ) {
+            if ( ! ((PropertyInfo) entry.getValue()).getClassInfo().equals(User.getOwnClassInfo()) ) {
+              logger.error("Cannot set a none User PropertyInfo as an email argument value");
+              continue;
+            }
+            String newValue = (String) ((PropertyInfo) entry.getValue()).get(user);
```

Comment:
Does this throw an exception if get( argument ) is the wrong class? Thinking if that was the case then you could just try/catch and forgo the explicit test above. 
Also, in the test above does it support Contact and Business?